app/operations/organizations/create.rb

```diff
@@ -11,11 +11,16 @@ class Create < Operation
     string :introduction, default: ''
     array :urls, default: []
     array :categories, default: []
+    array :tags, default: []
 
     def execute
       Organization.transaction(requires_new: true) do
         organization = build_organization
         organization.organization_contents.build(organization_contents_params)
+
+        tag_objects = Tags::BuildTags.run!(api_user: api_user, tag_array: tags) if tags
```

Comment:
What happens at the top level transaction in Organisations controller (create / update / destroy) when a `first_or_create` in the nested operation transaction fails here? Could we get inconsistent state?

http://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html

Reply:
This is a good question. I've added a spec to make sure that an error response is returned when the tag attribute is incorrectly formatted, but that doesn't answer the question. I think it might be moot because there's nothing happening in the top level transaction except the call to the operation; that one seems superfluous. 

Inside the op, using `require_new: true` should ensure that any exceptions in any of the sub-transactions (tag creation, saving objects, etc) should bubble up and cause the operation's transaction to roll back.

Reply:
the top level transaction will not rollback thus they could leave inconsistent state if there was a multi org update (via the array of orgs param) and one failed but the other didn't.

I'd have to double check if we support batch (array) creates, updates & deletes but that controller code looks like it does. Either way, I think we should add `require_new: true` to the top level to ensure a rollback on a nested transaction and thus keep the state consistent for the whole transaction. 

I think this PR is good to go so happy for this to come in another PR but we shouldn't allow the state to become inconsistent in the DB.